### DISCUSSION TITLE
A proper way to use next js for personal blog?

### DISCUSSION THREAD

**Discussion by HansLjy** [vercel/next.js#52171] [Help] [answered]
### Summary

I've seen some examples of blog powered by next js (like [this](https://vercel.com/templates/next.js/blog-starter-kit)). However, they commonly make the markdown files part of their repo, which is something I try to avoid. I do hope that the posts and the blog framework can be separated into two repos and whenever I pushed to the posts repo the blog framework will automatically regenerate all the pages. I tried to implement this by writing a script outside of both repos and run it on my server, but such solution is fairly unsatisfactory and error-prone. Is there any better way to do this?

**Comment by icyJoseph** [comment, marked as answer]
I use a [meilisearch](https://www.meilisearch.com/) DB, and send my posts to it. Then I have Next.js connect to that DB, and with [next-mdx-remote](https://github.com/hashicorp/next-mdx-remote) I render the page. 

Then with revalidate and when a request for a new post arrives, I connect again to the DB, update the content and move on.

**Reply by HansLjy** [reply]
Hi, thanks for your reply! So basically you render the page on client site instead of generating all static files before deploying?

**Reply by icyJoseph** [reply]
Nope, I do that on the build step. https://github.com/icyJoseph/icyJoseph/blob/dev/src/app/blog/%5Bslug%5D/page.tsx

**Reply by HansLjy** [reply]
Thanks. I realize that the function of revalidating is what I miss.
